code/game/machinery/bees_apiary.dm

```diff
@@ -103,180 +178,370 @@
 				to_chat(user, "<span class='notice'>You dislodge the apiary from the tray.</span>")
+
+			if (queen_bees_inside || worker_bees_inside)
+				empty_beehive()
+
+			for (var/datum/bee/B in bees_outside_hive)
+				B.home = null
+
 			qdel(src)
+
 	else if(istype(O, /obj/item/weapon/bee_net))
 		var/obj/item/weapon/bee_net/N = O
-		if(N.caught_bees > 0)
+		if(N.caught_bees.len)
 			to_chat(user, "<span class='notice'>You empty the bees into the apiary.</span>")
-			bees_in_hive += N.caught_bees
-			N.caught_bees = 0
+			for (var/datum/bee/B in N.caught_bees)
+				enterHive(B)
+			N.caught_bees = list()
 		else
 			to_chat(user, "<span class='notice'>There are no more bees in the net.</span>")
 	else
 		user.visible_message("<span class='warning'>\the [user] hits \the [src] with \the [O]!</span>","<span class='warning'>You hit \the [src] with \the [O]!</span>")
 		angry_swarm(user)
 
-/obj/machinery/apiary/Cross(atom/movable/mover, turf/target, height=1.5, air_group = 0)
-	if(air_group || (height==0))
-		return 1
-
-	if(istype(mover) && mover.checkpass(PASSTABLE))
-		return 1
+//Called every time a bee enters the hive.
+/obj/machinery/apiary/proc/enterHive(var/datum/bee/B)
+	bees_outside_hive.Remove(B)
+	if (istype(B,/datum/bee/queen_bee))
+		queen_bees_inside++
+		var/datum/bee/queen_bee/Q = B
+		if (Q.colonizing)
+			nutrilevel = max(15,nutrilevel+15)
 	else
+		worker_bees_inside++
+	B.home = src
+	B.state = null
+	B.health = B.maxHealth
+	for (var/datum/seed/S in B.pollens)
+		var/potency = round(S.potency)
+		var/totalreagents = 0
+		for(var/chem in S.chems)
+			var/list/reagent_data = list()
+			if (chem == NUTRIMENT)
+				if (S.products.len && juice_items.Find(S.products[1]))
+					reagent_data = S.chems[chem]
+				else
+					continue
+			else
+				reagent_data = S.chems[chem]
+			var/rtotal = reagent_data[1]
+			if(reagent_data.len > 1 && potency > 0)
+				rtotal += round(potency/reagent_data[2])
+			totalreagents += rtotal
+
+		if(totalreagents)
+			for(var/chem in S.chems)
+				var/list/reagent_data = list()
+				var/chemToAdd = null
+				if (chem == NUTRIMENT)
+					if (S.products.len && juice_items.Find(S.products[1]))
+						var/i = juice_items.Find(S.products[1])
+						chemToAdd = pick(juice_items[juice_items[i]])
+						reagent_data = S.chems[chem]
+					else
+						continue
+				else
+					reagent_data = S.chems[chem]
+					chemToAdd = chem
+				var/rtotal = reagent_data[1]
+				if(reagent_data.len > 1 && potency > 0)
+					rtotal += round(potency/reagent_data[2])
+				var/amountToAdd = min(1, max(0.25, rtotal/4))
+				var/difference = amountToAdd + 0.25 + reagents.total_volume - reagents.maximum_volume
+				if (difference>0)
+					reagents.trans_to(consume, difference)//This allows bees to bring in new reagents even if the hive is full
+				reagents.add_reagent(chemToAdd, amountToAdd * yieldmod)
+		if (!pollen.Find(S))
+			pollen.Add(S)
+		if (istype(B,/datum/bee/queen_bee))
+			reagents.add_reagent(ROYALJELLY,0.5 * yieldmod)
```

Comment:
Right, that's what I'm talking about in my initial post. The drawback is that you have to identify a queen and bring her with you in the scoop. It'd be nice if you could add royal jelly to an apiary so that you only have to scoop up workers.

Although this was roundabout to get you to understand what I was saying, I do think it was enlightening to have the concept of multiple queens elucidated more plainly, since that's not immediately apparent from your changelog. You can even use that bulletpoint list for the updated wiki page when this goes live.